### PR TITLE
Add warning text component

### PR DESCRIPTION
**Pull request by naseberry**
#### What
Add the Govuk Warning text component and replace existing markup

#### Ticket
[Add warning text component](https://dsdmoj.atlassian.net/browse/CBO-1388)

#### Why
Migrating to GDS design system to enable a consistent user experience

#### How
 - [X] write component spec
 - [X] write component helper
 - [X] replacing existing markup in views
 - [x] remove redundant styles

### CODE REVIEW

File: app/views/external_users/claims/_api_promo_banner.html.haml [hunk 1]
```diff
@@ -1,17 +1,12 @@
 - if claim.new_record? && show_api_promo_to_user?
 
   .js-callout-banner{'data-setting': 'api_promo_seen'}
-    .notice
-      %i.icon.icon-important
-        %span.visuallyhidden
-          = "Note"
+    = govuk_warning_text 'Do you use a case management system?', 'Note'
```
Comment (jsugarman) [inline app/views/external_users/claims/_api_promo_banner.html.haml:4]: **not a blocker** - translation

File: app/views/external_users/claims/_api_promo_banner.html.haml [hunk 2]
```diff
@@ -1,17 +1,12 @@
...
-        = "Do you use a case management system?"
     %p
       = "You may be able to import claims without re-typing. "
       = link_to 'Read more information.', api_landing_page_path
 
     %ul.list
       %li.hide-link
-        = link_to 'Do not show this message again', settings_user_path(current_user.id, api_promo_seen: 1), method: :put, remote: true
\ No newline at end of file
+        = link_to 'Do not show this message again', settings_user_path(current_user.id, api_promo_seen: 1), method: :put, remote: true
```
Comment (jsugarman) [inline app/views/external_users/claims/_api_promo_banner.html.haml:12]: **not a blocker** - translation

File: lib/govuk_component/warning_text_helpers.rb [hunk 3]
```diff
@@ -0,0 +1,16 @@
+# frozen_string_literal: true
+
+module GovukComponent
+  module WarningTextHelpers
```
Comment (jsugarman) [inline lib/govuk_component/warning_text_helpers.rb:4]: **not a blocker** - just in terms of naming - have you considered a "higher" level name that could be used to add other similar helpers - such as a for notice or alert type messages

**Review by jsugarman — APPROVED** [review]
👍 looks fine, but is the name too specific - is there a higher level of grouping here, e.g. GovukNoticeHelpers. On the other hand I see GDS have [isolated warning text ](https://design-system.service.gov.uk/components/warning-text/) as a "group" so not a blocker

File: lib/govuk_component/warning_text_helpers.rb (hunk 3 above)
Comment (naseberry) [reply lib/govuk_component/warning_text_helpers.rb:4]: I have had a similar discussion with design (Dan) as CCCD currently uses this component for warning / note / information.
We have decided to keep it consistent with the Govuk Design System and have further discussions later for a "higher" level name.

**Pull request merged**